ocpi-endpoints-msp-tokens/src/test/scala/com/thenewmotion/ocpi/tokens/MspTokensRouteSpec.scala

```diff
@@ -28,6 +27,29 @@ class MspTokensRouteSpec extends Specification with Specs2RouteTest with Mockito
         res.data mustEqual List(token)
       }
     }
+
+    "authorize with location references" in new TestScope {
+      service.authorize("23455655A", None) returns Future(\/-(AuthorizationInfo(Allowed.Allowed)))
+
+      Post("/23455655A/authorize") ~> route.route(apiUser) ~> check {
+        there was one(service).authorize("23455655A", None)
+        val res = entityAs[SuccessWithDataResp[AuthorizationInfo]]
+        res.data.allowed mustEqual Allowed.Allowed
+      }
+    }
+
+    "authorize without location references" in new TestScope {
+
+      val lr = LocationReferences("1234", List("1234"), List("1234", "5678"))
+
+      service.authorize("23455655A", Some(lr)) returns Future(\/-(AuthorizationInfo(Allowed.Allowed)))
+
+      Post("/23455655A/authorize", lr) ~> route.route(apiUser) ~> check {
+        there was one(service).authorize("23455655A", Some(lr))
+        val res = entityAs[SuccessWithDataResp[AuthorizationInfo]]
+        res.data.allowed mustEqual Allowed.Allowed
+      }
+    }
```

Comment:
It would be nice to see a test for when a request fails (with `MustProvideLocationReferences` for example).